ivy/functional/frontends/onnx/func_wrapper.py

```diff
@@ -0,0 +1,91 @@
+import functools
+from typing import Callable
+
+import ivy
+import ivy.functional.frontends.onnx as onnx_frontend
+
+
+def _from_ivy_array_to_onnx_frontend_tensor(x, nested=False, include_derived=None):
+    if nested:
+        return ivy.nested_map(
+            x, _from_ivy_array_to_onnx_frontend_tensor, include_derived, shallow=False
+        )
+    elif isinstance(x, ivy.Array) or ivy.is_native_array(x):
+        a = onnx_frontend.Tensor(x)
+        return a
+    return x
+
+
+def _ivy_array_to_onnx(x):
+    if isinstance(x, ivy.Array) or ivy.is_native_array(x):
+        return onnx_frontend.Tensor(x)
+    return x
+
+
+def _onnx_frontend_array_to_ivy(x):
+    if hasattr(x, "ivy_array"):
+        return x.ivy_array
+    return x
+
+
+def _native_to_ivy_array(x):
+    if isinstance(x, ivy.NativeArray):
+        return ivy.array(x)
+    return x
+
+
+def _to_ivy_array(x):
+    return _onnx_frontend_array_to_ivy(_native_to_ivy_array(x))
+
```

Comment:
I think defining ` _onnx_frontend_array_to_ivy` and `_native_to_ivy_array` just to use them here is a bit of an overkill 😅